src/thermo/PengRobinsonMFTP.cpp

```diff
@@ -0,0 +1,1321 @@
+						"unknown model ({}) or wrong number of parameters ({})",
+						iModel, vParams.size());
+				}
+			}
+		}
+}
+
+void PengRobinsonMFTP::setParametersFromXML(const XML_Node& thermoNode)
+{
+    MixtureFugacityTP::setParametersFromXML(thermoNode);
+    std::string model = thermoNode["model"];
+}
+
+doublereal PengRobinsonMFTP::sresid() const
+{
+    doublereal rho = density();
+    doublereal mmw = meanMolecularWeight();
+    doublereal molarV = mmw / rho;
+    double hh = m_b_current / molarV;
+    doublereal zz = z();
+    doublereal alpha_1 = daAlpha_dT();
+    doublereal T = temperature();
+    doublereal sqt2 = sqrt(2.0);
+		doublereal vpb = molarV + (1.0 + sqt2) *m_b_current;
+		doublereal vmb = molarV + (1.0 - sqt2) *m_b_current;
+		doublereal fac = alpha_1 / (2.0 * sqt2 * m_b_current);
+		doublereal sresid_mol_R = log(zz*(1.0 - hh)) + fac * log(vpb / vmb) / GasConstant;
+		return GasConstant * sresid_mol_R;
+}
+
+doublereal PengRobinsonMFTP::hresid() const
+{
+		doublereal rho = density();
+    doublereal mmw = meanMolecularWeight();
+    doublereal molarV = mmw / rho;
+    doublereal zz = z();
+		doublereal aAlpha_1 = daAlpha_dT();
+		doublereal T = temperature();
+		doublereal sqt2 = sqrt(2);
+		doublereal vpb = molarV + (1 + sqt2) *m_b_current;
+		doublereal vmb = molarV + (1 - sqt2) *m_b_current;
+		doublereal fac = 1 / (2.0 * sqt2 * m_b_current);
+		return GasConstant * T * (zz - 1.0) + fac * log(vpb / vmb) *(T * aAlpha_1 - m_aAlpha_current);
+}
+
+doublereal PengRobinsonMFTP::liquidVolEst(doublereal TKelvin, doublereal& presGuess) const
+{
+    double v = m_b_current * 1.1;
+    double atmp;
+    double btmp;
+		double aAlphatmp;
+		calculateAB(TKelvin, atmp, btmp, aAlphatmp);
+    doublereal pres = std::max(psatEst(TKelvin), presGuess);
+    double Vroot[3];
+    bool foundLiq = false;
+    int m = 0;
+    while (m < 100 && !foundLiq) {
+				int nsol = NicholsSolve(TKelvin, pres, atmp, btmp, aAlphatmp, Vroot);
+        if (nsol == 1 || nsol == 2) {
+            double pc = critPressure();
+            if (pres > pc) {
+                foundLiq = true;
+            }
+            pres *= 1.04;
+        } else {
+            foundLiq = true;
+        }
+    }
+
+    if (foundLiq) {
+        v = Vroot[0];
+        presGuess = pres;
+    } else {
+        v = -1.0;
+    }
+    return v;
+}
+
+doublereal PengRobinsonMFTP::densityCalc(doublereal TKelvin, doublereal presPa, int phaseRequested, doublereal rhoGuess)
+{
+    // It's necessary to set the temperature so that m_aAlpha_current is set correctly.
+    setTemperature(TKelvin);
+    double tcrit = critTemperature();
+    doublereal mmw = meanMolecularWeight();
+    if (rhoGuess == -1.0) {
+        if (phaseRequested != FLUID_GAS) {
+            if (TKelvin > tcrit) {
+                rhoGuess = presPa * mmw / (GasConstant * TKelvin);
+            } else {
+                if (phaseRequested == FLUID_GAS || phaseRequested == FLUID_SUPERCRIT) {
```

Comment:
This code is directly taken from the original 'RedlichKwongMFTP' file. However examining closely, I think that it should be FLUID_UNDEFINED instead of FLUID_GAS.

Reply:
If there is a bug, it should be fixed in both places, please

Reply:
These conditionals all set `rhoguess` to the same value, anyway, so I squashed them and eliminated some of the sorting, both here and in `RedlichKwongMFTP`,.  i.e., this is no longer an issue.  Marking resolved.